Update code to deploy Apache HTTP 2.4.39 on Ubuntu 16 Xenial stemcell using Bosh v2

sasadangelo opened the pull request:
Old Bosh lite v1 repository has been deprecated https://github.com/cloudfoundry/bosh-lite and CF move to Bosh v2 that has a different installation method: https://bosh.io/docs/bosh-lite/.
You can install bosh lite directly on your development machine using Virtualbox. The installation procedure creates a VM into the Virtual box where the bosh environment is created.

Bosh v2 changed all commands reported in the README.md file and for this reason I updated it.

Due to the new installation method script like install_prerequisites.sh are not required anymore.

Apache 2.4.39 installation requires the following libraries as prerequisites:
libapr1-dev
libaprutil1-dev
libpcre3-dev

you can build them from source code but I preferred to install them via apt-get install (see pre-start.sh) script.

As IP I used 10.243.1.2 instead of 10.244.1.2 because I had that IP already in use on my machine. Feel free to change it if you want.

I used Ubuntu 16 Xenial as stemcell because it was my requirement.

In the guide, I added the use of route command to redirect all traffic to 10.243.1.2 (or 10.244.1.2 if you want to change it) to 192.168.50.6.

httpd.conf.erb file didn't work with apache 2.4.39 so I had to change it.

georgethebeatle commented:
Hi @sasadangelo,

Thanks for your efforts! I think I managed to fix the conflicts. I will review and test your changes as soon as I can!

sasadangelo commented:
Hi @georgethebeatle,

Ok, thank you. FYI I tested it only on my Mac. I think without any change it should work on Linux as well, but I don't have time to do this kind of test. Feel free to apply any change to my code. Please let me know when you finished so I can merge with my fork.
Thank you again for your effort to do this package, it helped me a lot.

georgethebeatle commented:
Hi @sasadangelo,

I played around with your change. Overall looks good, but there is one little thing - you have chaged the app network address to `10.243.1.2`, but still `route add -net 10.244.0.0/16     192.168.50.6` which leads to the app not being accessible. You need to change the network address there as well. Is there a reason why 10.244 won't work?

Another thing is that the network configuration is no longer used in bosh v2 - these things are handled in the cloud config these days, but that can be later or as another PR. Otherwise there are deprecation messages during deployment.

Another minor thing is the packaging script - using `apt-get install` is generally discouraged in the bosh world. The correct thing would be to add more packages as dependencies. This thing as well can be addressed later.

sasadangelo commented:
Hi @georgethebeatle,

Yes, you're right about 10.244 vs 10.243. You can use 10.244. I changed it to 10.243 just because I started to study Bosh because I needed it for a project I am working on with my team about a PostgreSQL cluster with Patroni. I spent several days learning bosh because it was used by my team. Here the idea to use your project as an example for my learning.

However, on my development machine, 10.244 was already in use and I needed to change it. For this reason, I suggest you change back all reference to 10.243 in 10.244.

Regarding the network configuration, you're right. I noticed the warning but to be honest I ignored it because my focus was on other stuff. I can investigate how to solve this and create a new PR. I think you already have an idea on how to solve it, if so please give me more details.

Regarding the compilation vs apt-get install, you're right. My teammate informed me about this. To be honest I downloaded the libapr and libpre source code and built it successfully but at the end, I switched back to apt-get install for my purpose.

Compile apache with libapr, libaprutils and libpre is not hard. Read here:
https://stackoverflow.com/questions/9436860/apache-httpd-setup-and-installation. 

Simply download apr and apr-util from here:
http://apr.apache.org/

modify the spec file to add them. In the packaging file, extract them and move in srclib folder (a folder you must create in the apache root folder). You need also to build the libpre libraries. Read here on how to build it:
https://stackoverflow.com/questions/10663180/apache-installation-libpcre-error.

However, if it is ok for you I can create a separate PR for this.

georgethebeatle commented:
I moved back to 10.244.1.0 and ,erged as #9 (I had a problem updating this one). I would approceiate it if you PR the proper packaging of the apache server.

sasadangelo commented:
Hi @georgethebeatle,

Thank you for merging. I realigned my fork and this evening (max tomorrow) Rome time I'll try to remove the apt-get install in favor of apache compilation. I'll create a PR as soon as the code will be ready. It's ok for me to have 10.244.x.x as said before. In the next PR I will make sure to not change it.

Regarding the removal of network warning, I'll try to fix it as well. I'll let you know progresses.

Have a nice day.

georgethebeatle commented:
Cool, thanks a lot @sasadangelo!
